Tools/i18n/pygettext.py

```diff
@@ -343,6 +344,29 @@ def __waiting(self, ttype, tstring, lineno):
                 return
         if ttype == tokenize.NAME and tstring in opts.keywords:
             self.__state = self.__keywordseen
+            return
+        if ttype == tokenize.STRING:
+            maybe_fstring = ast.parse(tstring, mode='eval').body
+            if not isinstance(maybe_fstring, ast.JoinedStr):
+                return
+            for value in filter(lambda node: isinstance(node, ast.FormattedValue),
+                                maybe_fstring.values):
+                for call in filter(lambda node: isinstance(node, ast.Call),
+                                   ast.walk(value)):
+                    func = call.func
+                    # Name.id or Attribute.attr
+                    func_name = getattr(func, "id", None) or func.attr
+                    if func_name not in opts.keywords:
+                        continue
```

Comment:
For an input such as;
```py
foo = f'{type(str)("foo bar baz")}'
```
it will fail since it assumes all calls inside of f-strings have their function as either attribute or a name node. Probably something like this solves;
```suggestion
                    if isinstance(func, ast.Name):
                        func_name = func.id
                    elif isinstance(func, ast.Attribute):
                        func_name = func.attr
                    else:
                        continue
```